src/fn_call.rs

```diff
@@ -71,8 +76,196 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
 
         // First: functions that could diverge.
         match link_name {
-            "__rust_start_panic" | "panic_impl" => {
-                return err!(MachineError("the evaluated program panicked".to_string()));
+            "panic_impl" => {
+                // Manually forward to 'panic_impl' lang item
+                let panic_impl_real = this.tcx.lang_items().panic_impl().unwrap();
+
+                return Ok(Some(this.load_mir(InstanceDef::Item(panic_impl_real))?));
```

Comment:
I'd prefer if you could move this out of `emulate_foreign_item`... then you can also keep the old signature.

Instead, maybe add a new method `forward_lang_item` or so?

Reply:
This comment still stands. `emulate_foreign_item` should always proceed by emulation, not by finding MIR.